### PR TITLE
Bug 1127547 - Follow-up fixes for custom context menu

### PR DESCRIPTION
**Pull request by thebnich**
Follow-up to #61. I had suggested we try making the gesture recognizer implement the BrowserHelper protocol, but that doesn't seem to buy us anything and just adds extra code, so I removed it. Also fixes the JS race by disabling only the built-in context menu gesture recognizer.

### CODE REVIEW

**Comment by st3fan** [comment]
With regard to namespacing .. can we do a similar thing as in https://github.com/mozilla/firefox-ios/pull/116/files#diff-2 ? I am also going to follow that pattern for the ReaderMode.js code.

File: Client/Frontend/Browser/LongPressGestureRecognizer.swift [hunk 1]
```diff
@@ -43,64 +42,41 @@ class LongPressGestureRecognizer: UILongPressGestureRecognizer, UIGestureRecogni
...
-            touchLocation.y /= self.browser.webView.scrollView.zoomScale
-
-            self.browser.webView.evaluateJavaScript("findElementsAtPoint(\(touchLocation.x),\(touchLocation.y))", completionHandler:nil)
+            var touchLocation = gestureRecognizer.locationInView(self.webView)
+            touchLocation.x -= self.webView.scrollView.contentInset.left
+            touchLocation.y -= self.webView.scrollView.contentInset.top
+            touchLocation.x /= self.webView.scrollView.zoomScale
+            touchLocation.y /= self.webView.scrollView.zoomScale
+
+            self.webView.evaluateJavaScript("findElementsAtPoint(\(touchLocation.x),\(touchLocation.y))") { (response: AnyObject!, error: NSError!) in
```
Comment (st3fan) [inline Client/Frontend/Browser/LongPressGestureRecognizer.swift]: How reliable is this? Is there no alternative way to do this?

Comment (thebnich) [inline Client/Frontend/Browser/LongPressGestureRecognizer.swift]: I can't think of another way. The gesture recognizer works in UIView coordinates, and we need to give page coordinates to the JS.

Comment (st3fan) [inline Client/Frontend/Browser/LongPressGestureRecognizer.swift]: I understand. Some time in the future we should experiment with doing the long tap detection in JS I think.

**Comment by thebnich** [comment]
Merged 4219726.

**Pull request closed without merging**